src/compiler/scala/tools/nsc/transform/ExplicitOuter.scala

```diff
@@ -447,9 +449,10 @@ abstract class ExplicitOuter extends InfoTransform
           if (sym.isProtected && //(4)
               (qsym.isTrait || !(qual.isInstanceOf[Super] || (qsym isSubClass currentClass))))
             sym setFlag notPROTECTED
-          super.transform(tree)
+          if (sym.isStaticMember) tree
+          else super.transform(tree)
 
-        case Apply(sel @ Select(qual, nme.CONSTRUCTOR), args) if isInner(sel.symbol.owner) =>
+        case treeInfo.Applied(sel @ Select(qual, name), _, List(args)) if needsOuterParam(sel.symbol) =>
```

Comment:
Applied is a dangerous extractor, I prefer to avoid it -- from its docs:
> Trees which are not applications are also accepted. Their callee and core will
> be equal to the input, while targs and argss will be Nil.

Can we reformulate this case without it? Was it a bug that we weren't considering `TypeApply`s?

Reply:
I guess in this case you are ensuring args is non-empty, so we're good, but still I always worry when I see this pattern

Reply:
Ah... yup it was to catch applications with and without `TypeApply`.